benches/main.rs

```diff
@@ -0,0 +1,133 @@
+            assert_eq!(ev_count, no_of_subscribers)
+        })
+    });
+}
+
+// Test the performance of event manager when the subscribers are wrapped in an Arc, and they
+// leverage inner mutability to update their internal state.
+// The performance is assessed under stress, all added subscribers have active events.
+fn run_subscriber_with_inner_mut(c: &mut Criterion) {
+    let no_of_subscribers = 200;
+
+    let mut event_manager = EventManager::<Arc<dyn EventSubscriber + Send + Sync>>::new().unwrap();
+    for _ in 0..no_of_subscribers {
+        let counter_subscriber = CounterInnerMutSubscriber::new();
+        counter_subscriber.trigger_event();
+        event_manager.add_subscriber(Arc::new(counter_subscriber));
+    }
+
+    c.bench_function("process_with_inner_mut", |b| {
+        b.iter(|| {
+            let ev_count = event_manager.run().unwrap();
+            assert_eq!(ev_count, no_of_subscribers)
+        })
+    });
+}
+
+// Test the performance of event manager when it manages subscribers of different types, that are
+// wrapped in an Arc<Mutex>. Also, make use of `Events` with custom user data
+// (using CounterSubscriberWithData).
+// The performance is assessed under stress, all added subscribers have active events, and the
+// CounterSubscriberWithData subscribers have multiple active events.
+fn run_multiple_subscriber_types(c: &mut Criterion) {
+    let no_of_subscribers = 100;
+
+    let mut event_manager = EventManager::<Arc<Mutex<dyn MutEventSubscriber>>>::new()
+        .expect("Cannot create event manager.");
+
+    for i in 0..no_of_subscribers {
+        // The `CounterSubscriberWithData` expects to receive a number as a parameter that
+        // represents the number it can use as its inner Events data.
+        let mut data_subscriber = CounterSubscriberWithData::new(i * no_of_subscribers);
+        data_subscriber.trigger_all_counters();
+        event_manager.add_subscriber(Arc::new(Mutex::new(data_subscriber)));
+
+        let mut counter_subscriber = CounterSubscriber::new();
+        counter_subscriber.trigger_event();
+        event_manager.add_subscriber(Arc::new(Mutex::new(counter_subscriber)));
+    }
+
+    c.bench_function("process_dynamic_dispatch", |b| {
+        b.iter(|| {
+            let _ = event_manager.run().unwrap();
+        })
+    });
+}
+
+// Test the performance of event manager when it manages a single subscriber type.
+// Just a few of the events are active in this test scenario.
+fn run_with_few_active_events(c: &mut Criterion) {
+    let no_of_subscribers = 200;
+
+    let mut event_manager = EventManager::<CounterSubscriber>::new().unwrap();
+
+    for i in 0..no_of_subscribers {
+        let mut counter_subscriber = CounterSubscriber::new();
+        // Let's activate the events for a few subscribers (i.e. only the ones that are
+        // divisible by 23). 23 is a random number that I just happen to like.
+        if i % 23 == 0 {
```

Comment:
We could pick a number `n` of active subscribers, and then do an `if i < n { counter_subscriber.trigger_event(); }` maybe.

Reply:
But... 23 :(